### PR TITLE
Preload Refactor

### PR DESCRIPTION
**Pull request by jwerle**
Imports `socket:init` in preload script for global setup, like managing the concurrency of `ipc://post` XHR requests. Heavy usage makes the webview freeze and these are uninitiated requests, so we should throttle them. This PR also makes the runtime work in web workers

### CODE REVIEW

**Comment by chicoxyzzy** [comment]
hmm, maybe we can initialize primordials in the same phase?

**Comment by chicoxyzzy** [comment]
and FS constants

**Comment by jwerle** [comment]
> hmm, maybe we can initialize primordials in the same phase?

they are with `import ipc from 'socket:ipc'`

**Comment by jwerle** [comment]
@chicoxyzzy this one is ready to go!

File: src/core/core.cc [hunk 1]
```diff
@@ -109,41 +109,28 @@ namespace SSC {
...
-      "xhr.send();                                                   \n"
+      "const globals = await import('socket:internal/globals');              \n"
+      "const id = `" + sid + "`;                                             \n"
+      "const seq = `" + seq + "`;                                            \n"
+      "const headers = `" + trim(post.headers) + "`                          \n"
+      "  .trim()                                                             \n"
+      "  .split(/[\\r\\n]+/)                                                 \n"
+      "  .filter(Boolean);                                                   \n"
+      "                                                                      \n"
+      "let params = `" + params + "`;                                        \n"
```
Comment (heapwolf) [inline src/core/core.cc:120]: does this need to escape backticks?

**Review by heapwolf — APPROVED** [review]

File: src/core/core.cc (hunk 1 above)
Comment (jwerle) [reply src/core/core.cc:120]: it shouldn't! I just copied this from what was there before

**Review by mribbons — CHANGES_REQUESTED** [review]
I know it's not related to this branch specifically, but:
```cli.cc:768:10: error: use of undeclared identifier 'SIGUSR1'```

Also, fix linting :)

**Review by chicoxyzzy — APPROVED** [review]
LGTM

**Pull request merged**